### DISCUSSION TITLE
vpc with nlmixr2

### DISCUSSION THREAD

**Discussion by jiankelius** [nlmixr2/nlmixr2#291] [Q&A]
Hello,

try to do vpc with nlmixr2::vpcPlot(), but got the following error message:
Error in sim$sim[sim[[id]] == i] <- cumsum(stats::ave(1:length(sim_id[sim[[id]] ==  : 
  NAs are not allowed in subscripted assignments
I saw an earlier discussion on this, but could not resolve the issue based on the discussion. 

Therefore, decide to use tidyvpc. to use tidyvpc, first I need to simulate replicates of data with simulate(). However, I could not find documentation on simulate() and do not understand its output:
is cp same as DV, ipredSim same as IPRED and sim same as PRED?

Appreciate if someone can point me the documentation of simulate(). Also appreciate if a better way of doing vpc, please inform.

thanks in advance,

Jack

**Comment by mattfidler** [comment]
Hi @jiankelius 

I would suggest using `vpcSim`:

https://nlmixr2.github.io/nlmixr2est/reference/vpcSim.html

**Reply by jiankelius** [reply]
Thanks, Matt for your prompt response.

Below are the codes when using tidyvpc:
vpc <- observed(obs_data, x=TIME, y=DV) %>%
  simulated(sim_data, y=DV) %>%
  binning(bin = "jenks", nbins = 5) %>%
  vpcstats()

which column of vpcSim output (I see ipred, sim and pred) is to be used for DV in "simulated(sim_data, y=DV)" above?

**Reply by mattfidler** [reply]
See examples here:

https://certara.github.io/tidyvpc/articles/tidyvpc_nlmixr2.html
